web/src/client/network.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) [2022] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import { applyMixin, withDBus } from "./mixins";
+
+const NM_IFACE = "org.freedesktop.NetworkManager";
+
+export default class NetworkClient {
+  constructor(dbusClient) {
+    this._client = dbusClient;
+  }
+
+  /**
+   * Returns IP config overview - addresses and hostname
+   *
+   * @return {Promise.<Map>} address key stores list of addresses,
+   *                         hostname stores target's hostname
+   */
+  async config() {
+    const data = await this.#addresses();
+    const addrs = data.map(a => {
+      return {
+        address: a.address.v,
+        prefix: a.prefix.v
+      };
+    });
+
+    return {
+      addresses: addrs,
```

Comment:
Hint: actually you can use `const addresses = data.map....` above and then use directly here as below

```suggestion
      addresses,
```

See https://javascript.info/object#property-value-shorthand